raft/raft.hh

```diff
@@ -450,12 +442,31 @@ struct entry_id {
     index_t idx;
 };
 
+// The execute_add_entry/execute_modify_config methods can return this error to signal
+// that the request should be retried after a delay.
```

Comment:
But the patch also replaces not_a_leader with this exception. Those need to be retried not after delay but after looking up a new leader. In fact I do not see what condition requires delay before retrying `add_entry`.

Reply:
If stepdown is in progress (https://github.com/scylladb/scylladb/blob/master/raft/fsm.cc#L67), we may not yet know the new leader, if I understand it right, and we will keep retrying in a hot cycle until the informatation about the new leader reaches the client node.